flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -86,32 +144,25 @@ public KinesisProxy(Properties configProps) {
 	 * @param maxRecordsToGet the maximum amount of records to retrieve for this batch
 	 * @return the batch of retrieved records
 	 */
-	public GetRecordsResult getRecords(String shardIterator, int maxRecordsToGet) {
+	@Override
+	public GetRecordsResult getRecords(String shardIterator, int maxRecordsToGet) throws InterruptedException {
 		final GetRecordsRequest getRecordsRequest = new GetRecordsRequest();
 		getRecordsRequest.setShardIterator(shardIterator);
 		getRecordsRequest.setLimit(maxRecordsToGet);
 
 		GetRecordsResult getRecordsResult = null;
 
-		int remainingRetryTimes = Integer.valueOf(
-			configProps.getProperty(KinesisConfigConstants.CONFIG_STREAM_DESCRIBE_RETRIES, Integer.toString(KinesisConfigConstants.DEFAULT_STREAM_DESCRIBE_RETRY_TIMES)));
-		long describeStreamBackoffTimeInMillis = Long.valueOf(
-			configProps.getProperty(KinesisConfigConstants.CONFIG_STREAM_DESCRIBE_BACKOFF, Long.toString(KinesisConfigConstants.DEFAULT_STREAM_DESCRIBE_BACKOFF)));
-
-		int i=0;
-		while (i <= remainingRetryTimes && getRecordsResult == null) {
+		int attempt = 0;
+		while (attempt <= getRecordsMaxAttempts && getRecordsResult == null) {
 			try {
 				getRecordsResult = kinesisClient.getRecords(getRecordsRequest);
 			} catch (ProvisionedThroughputExceededException ex) {
+				long backoffMillis = fullJitterBackoff(
+					getRecordsBaseBackoffMillis, getRecordsMaxBackoffMillis, getRecordsExpConstant, attempt++);
 				LOG.warn("Got ProvisionedThroughputExceededException. Backing off for "
-					+ describeStreamBackoffTimeInMillis + " millis.");
-				try {
-					Thread.sleep(describeStreamBackoffTimeInMillis);
-				} catch (InterruptedException interruptEx) {
-					//
-				}
+					+ backoffMillis + " millis.");
+				Thread.sleep(backoffMillis);
 			}
-			i++;
 		}
 
 		if (getRecordsResult == null) {
```

Comment:
Not sure if throwing an exception here is a good idea.
From the javadocs of `getRecord()` it says: "Note that if the shard has been closed, the shard iterator can't return more data and <a>GetRecords</a> returns <code>null</code>". This means that our code would fail each time some shards are closed?